File: 진우/5weeks/1157. 단어 공부/단어 공부.py
word = input().strip().upper()  # 입력받고 대문자로 변환

alpa = [0] * 26  # 알파벳 개수 저장용 리스트

# 각 알파벳 카운트
for c in word:
    alpa[ord(c) - ord('A')] += 1

# 최댓값 찾기
mx = max(alpa)
count = alpa.count(mx)

# 결과 출력
if count > 1:
    print("?")
else:
    result = chr(alpa.index(mx) + ord('A'))
    print(result)

File: 진우/5weeks/1157. 단어 공부/단어 공부.cc
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
#include <cctype> // ::toupper

using namespace std;

int main() {
    string word;
    cin >> word;

    vector<int> alpa(26, 0);

    // 대문자로 변환
    transform(word.begin(), word.end(), word.begin(), ::toupper);

    // 각 알파벳 카운트
    for (char c : word) {
        alpa[c - 'A']++;
    }

    // 최댓값 찾기
    int mx = *max_element(alpa.begin(), alpa.end());
    int count = 0;
    char result;

    // 최댓값이 몇 개 있는지 확인
    for (int i = 0; i < 26; i++) {
        if (alpa[i] == mx) {
            count++;
            result = 'A' + i;
        }
    }

    // 출력
    if (count > 1) cout << "?" << endl;
    else cout << result << endl;

    return 0;
}
